stm32/Src/phytos31.c

```diff
@@ -0,0 +1,45 @@
+/**
+ ******************************************************************************
+ * @file    sdi12.c
+ * @author  Stephen Taylor
+ * 
+ * @brief   This library is designed to read measurements from a PHYTOS-31 
+ *          sensor from METER.
+ *          https://metergroup.com/products/phytos-31/
+ * @date    4/18/2024
+ ******************************************************************************
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "phytos31.h"
+#include "transcoder.h"
+
+HAL_StatusTypeDef Phytos31Init(){
+  return ADC_init();
+}
+
+phytos_measurments Phytos31GetMeasurment(){
```

Comment:
not currently, can be integrated

Reply:
I think that would be good to add so that data with errors won't be added to the buffer. That is a bug that I don't want to have to track down. In `Phytos31_measure` you can return `-1` as an error indicator and it won't be added to the buffer.